core/vm/analysis.go

```diff
@@ -41,21 +41,47 @@ func (d destinations) has(codehash common.Hash, code []byte, dest *big.Int) bool
 		m = jumpdests(code)
 		d[codehash] = m
 	}
-	return (m[udest/8] & (1 << (udest % 8))) != 0
+	return OpCode(code[udest]) == JUMPDEST && (m[udest/8]&(0x80>>(udest%8))) == 0
+	//	return (m[udest/8] & (1 << (udest % 8))) != 0
+}
+
+type bitvec struct {
+	m []byte
+}
```

Comment:
Please add a comment about the layout of bitvec, i.e. explain what's in the slice.